code/datums/spells/infinite_guns.dm

```diff
@@ -17,5 +19,41 @@
 		C.drop_item()
 		C.swap_hand()
 		C.drop_item()
-		var/obj/item/gun/projectile/shotgun/boltaction/enchanted/GUN = new
+		var/obj/item/gun/projectile/shotgun/boltaction/enchanted/GUN = new gun_type()
 		C.put_in_hands(GUN)
+
+/obj/effect/proc_holder/spell/infinite_guns/fireball
+	name = "Rapid-fire Fireball"
+	desc = "Multiple Fireballs. Need I explain more? Requires both hands free to use."
+	invocation_type = "none"
+
+	school = "evocation"
+	base_cooldown = 30 SECONDS
+	clothes_req = FALSE
+	invocation = "ONI SOMA-SOMA-SOMA"
+	invocation_type = "shout"
+
+	action_icon_state = "explosion"
+
+	gun_type = /obj/item/gun/projectile/shotgun/boltaction/enchanted/arcane_barrage/fireball
+
+/obj/item/gun/projectile/shotgun/boltaction/enchanted/arcane_barrage/fireball
+	name = "small ball of fire"
+	desc = "A small flame, ready to launch from your hand."
+	icon = 'icons/obj/cigarettes.dmi'
+	icon_state = "match_unathi"
+	item_state = "disintegrate"
+	lefthand_file = 'icons/mob/inhands/items_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/items_righthand.dmi'
+	color = "#e1ff00" // red + yellow = orange
+	guns_left = 20
+	fire_sound = 'sound/magic/Fireball.ogg'
+	mag_type = /obj/item/ammo_box/magazine/internal/boltaction/enchanted/arcane_barrage/fireball
+	flags = NOBLUDGEON | DROPDEL
```

Comment:
```suggestion
```
already declared

Reply:
but they're not tho
![image](https://github.com/ParadiseSS13/Paradise/assets/91113370/4227d3cb-fc99-406d-9b47-8cd092786583)

Reply:
Didn't see that, feel free to resolve